cid/helpers/cur_proxy.py

```diff
@@ -0,0 +1,452 @@
+                        map_mapping[cur1_field[len(field + '_'):]] = cur1_field
+                if not map_mapping:
+                    return 'cast(NULL AS MAP<VARCHAR, VARCHAR>)'
+                return f'''
+                    MAP(
+                        ARRAY[{', '.join(["'" + key + "'" for key in map_mapping.keys()])}],
+                        ARRAY[{', '.join([cur1_field for cur1_field in map_mapping.values()])}]
+                    )
+                '''
+            cur1to2mapping = {value: key for key, value in cur2to1mapping.items()}
+            if field in cur1to2mapping:
+                return f'{cur1to2mapping.ge(field, field)}'
+            else:
+                raise NotImplementedError(f'WARNING: {field} has not known equivalent')
+
+        if self.current_cur_version.startswith('2') and self.target_cur_version.startswith('1'):
+            if field.startswith('resource_tags_'):
+                return f"resource_tags['{field[len('resource_tags_'):]}']"
+            if field.startswith('cost_category_'):
+                return f"cost_category['{field[len('cost_category_'):]}']"
+            return cur2to1mapping.get(field, field)
+
+    def create_or_update_view(self):
+        self.read_from_athena()
+        all_fields  = list(set(self.exposed_fields + self.fields_to_expose))
+        lines = []
+        print('all_fields',  all_fields)
+        for field in all_fields:
+            field_type = self.cur.get_type_of_column(field)
+            mapped_expression = self.get_sql_expression(field, field_type)
+            requirement = mapped_expression.split('[')[0]
+            if not re.match(r'^[a-zA-Z0-9_]+$', requirement) or self.cur.column_exists(requirement):
+                expression = mapped_expression
+            else:
+                if field_type not in empty:
+                    raise RuntimeError(f'{field_type} not in empty list')
+                expression = empty.get(field_type, 'null')
+            lines.append(f'{expression} {field}')
+        select_block = '\n                ,'.join(lines)
+        query = (f'''
+            CREATE OR REPLACE VIEW "{self.name}" AS
+            SELECT
+                {select_block}
+            FROM
+                "{self.cur.table_name}"
+        ''')
+
+        logging.critical(query)
+        res = self.athena.query(query)
+        logging.debug(res)
+
+    def get_table_metadata(self):
+        return self.athena.get_table_metadata(self.name)
+
+
+if __name__ == '__main__':
+
+    import boto3
+    from cid.helpers.athena import Athena
+    from cid.helpers.glue import Glue
+    from cid.helpers.cur import CUR
+
+    logging.basicConfig(level=logging.INFO)
+    logger.setLevel(logging.DEBUG)
+    athena = Athena(session=boto3.Session())
+    glue = Glue(session=boto3.Session())
+    cur = CUR(athena=athena, glue=glue)
+    proxy = ProxyView(
+        cur=cur,
+        target_cur_version='2',
+        fields_to_expose = {
```

Comment:
What purpose does this serve? And why we hard code this here, but at line 341 we are joining this with default columns? Can't we just add them to default_columns? 

self.fields_to_expose = list(set(default_columns[self.target_cur_version] + (fields_to_expose or []) ))

Reply:
This is a test code.